Dockerfile

```diff
@@ -54,6 +54,13 @@ RUN --mount=type=cache,target=/root/.cache/go-build \
     go build -trimpath -ldflags "${ldflags} -extldflags '-static'" \
     -o manager ${package}
 
+# Build the clusterctl binary
+RUN --mount=type=cache,target=/root/.cache/go-build \
```

Comment:
Will it be okay to make this a separate RUN or chain the go build in a single command?

Reply:
I think it should be possible to pass the build argument with clusterctl package `package=./cmd/clusterctl` and make the output binary an argument with a default to `manager` like this:
```
ARG outbin=manager
```
Then in Makefile you can:
```
DOCKER_BUILDKIT=1 docker build --build-arg builder_image=$(GO_CONTAINER_IMAGE) --build-arg goproxy=$(GOPROXY) --build-arg ARCH=$(ARCH) --build-arg package=./cmd/clusterctl --build-arg outbin=clusterctl --build-arg ldflags="$(LDFLAGS)" . -t $(CLUSTERCTL_IMG)-$(ARCH):$(TAG)
```

This way we will avoid including any other binaries in the clusterctl image.

Reply:
That's great! Thanks. 
Also, do we have to make changes to push the image to `gcr.io` or will that be done as part of the release?

Reply:
I think if you add the push target accordingly it should be fine. This will publish to the staging registry and promotion to gcr.io during a release is done semi-manually via image promotion.